Many unittests broken:

```console
ERROR tests/test_plotly_utils/validators/test_pandas_series_input.py::test_color_validator_categorical[Index] - RecursionError: maximum recursion depth exceeded
= 175 failed, 2855 passed, 21 skipped, 192 deselected, 4 xfailed, 686 warnings, 2 errors in 74.32s (0:01:14) =
```

error:

```console
tests/test_plotly_utils/validators/test_xarray_input.py:126: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
_plotly_utils/basevalidators.py:1395: in validate_coerce
    self.raise_invalid_val(v)
_plotly_utils/basevalidators.py:308: in raise_invalid_val
    v=repr(v),
/nix/store/qzqlix3rhk63zsmf05p1w14cx78rh96c-python3.12-xarray-2025.04.0/lib/python3.12/site-packages/xarray/core/common.py:184: in>
    return formatting.array_repr(self)
/nix/store/q54d9qqlb27g2nf9ja4zsa89gl7b3yvm-python3-3.12.11/lib/python3.12/reprlib.py:21: in wrapper
    result = user_function(self)
/nix/store/qzqlix3rhk63zsmf05p1w14cx78rh96c-python3.12-xarray-2025.04.0/lib/python3.12/site-packages/xarray/core/formatting.py:688>
    data_repr = short_data_repr(arr)
/nix/store/qzqlix3rhk63zsmf05p1w14cx78rh96c-python3.12-xarray-2025.04.0/lib/python3.12/site-packages/xarray/core/formatting.py:657>
    return limit_lines(repr(array.data), limit=40)
/nix/store/4xvvphpbvk5kkiqqnq3hflipbzb7ym2q-python3.12-numpy-2.3.0/lib/python3.12/site-packages/numpy/_core/arrayprint.py:1621: in>
    extras.append(f"dtype={dtype_short_repr(arr.dtype)}")
/nix/store/4xvvphpbvk5kkiqqnq3hflipbzb7ym2q-python3.12-numpy-2.3.0/lib/python3.12/site-packages/numpy/_core/arrayprint.py:1574: in>
    return f"'{str(dtype)}'"
/nix/store/4xvvphpbvk5kkiqqnq3hflipbzb7ym2q-python3.12-numpy-2.3.0/lib/python3.12/site-packages/numpy/__init__.py:852: in <module>
    from . import exceptions
<frozen importlib._bootstrap>:1412: in _handle_fromlist
    ???
/nix/store/4xvvphpbvk5kkiqqnq3hflipbzb7ym2q-python3.12-numpy-2.3.0/lib/python3.12/site-packages/numpy/__init__.py:733: in __getatt>
    import numpy.exceptions as exceptions
/nix/store/4xvvphpbvk5kkiqqnq3hflipbzb7ym2q-python3.12-numpy-2.3.0/lib/python3.12/site-packages/numpy/__init__.py:733: in __getatt>
    import numpy.exceptions as exceptions
E   RecursionError: maximum recursion depth exceeded
!!! Recursion detected (same locals & position)
```

caused by the following code:

```python
"""
Stand-alone module to provide information about whether optional deps exist.

"""

from importlib import import_module
import logging
import sys

logger = logging.getLogger(__name__)
_not_importable = set()


def get_module(name, should_load=True):
    """
    Return module or None. Absolute import is required.

    :param (str) name: Dot-separated module path. E.g., 'scipy.stats'.
    :raise: (ImportError) Only when exc_msg is defined.
    :return: (module|None) If import succeeds, the module will be returned.

    """
    if not should_load:
        return sys.modules.get(name, None)

    if name not in _not_importable:
        try:
            return import_module(name)
        except ImportError:
            _not_importable.add(name)
        except Exception:
            _not_importable.add(name)
            msg = f"Error importing optional module {name}"
            logger.exception(msg)

    return None
```

--- a/pkgs/development/python-modules/plotly/default.nix
+++ b/pkgs/development/python-modules/plotly/default.nix
@@ -103,11 +103,28 @@ buildPythonPackage rec {
     "test_acceptance_named"
   ];
 
+  __darwinAllowLocalNetworking = true;
+
   disabledTestPaths = lib.optionals stdenv.hostPlatform.isDarwin [
-    # requires local networking
-    "plotly/tests/test_io/test_renderers.py"
     # fails to launch kaleido subprocess
-    "plotly/tests/test_optional/test_kaleido"
+    "tests/test_optional/test_kaleido"
+    # numpy2 related error, RecursionError
+    # See: https://github.com/plotly/plotly.py/issues/4852
+    "tests/test_plotly_utils/validators/test_angle_validator.py"
+    "tests/test_plotly_utils/validators/test_any_validator.py"
+    "tests/test_plotly_utils/validators/test_color_validator.py"
+    "tests/test_plotly_utils/validators/test_colorlist_validator.py"
+    "tests/test_plotly_utils/validators/test_colorscale_validator.py"
+    "tests/test_plotly_utils/validators/test_dataarray_validator.py"
+    "tests/test_plotly_utils/validators/test_enumerated_validator.py"
+    "tests/test_plotly_utils/validators/test_fig_deepcopy.py"
+    "tests/test_plotly_utils/validators/test_flaglist_validator.py"
+    "tests/test_plotly_utils/validators/test_infoarray_validator.py"
+    "tests/test_plotly_utils/validators/test_integer_validator.py"
+    "tests/test_plotly_utils/validators/test_number_validator.py"
+    "tests/test_plotly_utils/validators/test_pandas_series_input.py"
+    "tests/test_plotly_utils/validators/test_string_validator.py"
+    "tests/test_plotly_utils/validators/test_xarray_input.py"
   ];
 
   pythonImportsCheck = [ "plotly" ];

--- a/pkgs/development/python-modules/plotly/default.nix
+++ b/pkgs/development/python-modules/plotly/default.nix
@@ -38,14 +38,14 @@
 
 buildPythonPackage rec {
   pname = "plotly";
-  version = "6.1.0";
+  version = "6.1.2";
   pyproject = true;
 
   src = fetchFromGitHub {
     owner = "plotly";
     repo = "plotly.py";
     tag = "v${version}";
-    hash = "sha256-B5wjZTnL/T+zRbPd3tVSekDbYnKBvIdIpXhc3sUvT3E=";
+    hash = "sha256-+vIq//pDLaaTmRGW+oytho3TfMmLCtuIoHeFenLVcek=";
   };
 
   postPatch = ''

